reactnative_mobile/react_app_todo_list/App.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {useState} from 'react';
+import { KeyboardAvoidingView, StyleSheet, Text, View, TextInput, TouchableOpacity, Keyboard, ScrollView, Platform } from 'react-native';
+import Task from './components/Task';
+import { any } from 'prop-types';
+
+export default function App() {
+  
+  const [task, setTask] = useState<any>();
+  const [taskItems, setTaskItems] = useState<any[]>([]);
+
+  const handleAddTask = () => {
+    Keyboard.dismiss();
+    setTaskItems([...taskItems, task])
+    setTask(null);
+  }
+
+  const completeTask = (index: any) => {
+    let itemsCopy = [...taskItems];
+    itemsCopy.splice(index, 1);
+    setTaskItems(itemsCopy)
+  }
+
+  return (
+    <View style={styles.container}>
+      {/* Added this scroll view to enable scrolling when list gets longer than the page */}
```

Comment:
This enables scrolling, but I believe it will render all items at once. This will be a perf issue if the list grows long. Please use a FlatList.

See https://reactnative.dev/docs/using-a-listview

> Unlike the more generic [ScrollView](https://reactnative.dev/docs/using-a-scrollview), the FlatList only renders elements that are currently showing on the screen

Reply:
Okay will make this change.